mu/interface/dialogs.py

```diff
@@ -460,17 +454,17 @@ def update_firmware(self):
         if device is None:
             return
 
-        esptool = "-mesptool"
+        esptool = "-mmu.contrib.esptool"
         erase_command = '"{}" "{}" --port {} erase_flash'.format(
-            venv.interpreter, esptool, device.port
+            sys.executable, esptool, device.port
         )
 
         if self.device_type.currentText() == "ESP32":
             write_command = (
                 '"{}" "{}" --chip esp32 --port {} --baud {} '
                 'write_flash -z 0x1000 "{}"'
             ).format(
-                venv.interpreter,
+                sys.executable,
```

Comment:
Why the change from `venv.interpreter` to `sys.executable`..?

Reply:
esptool used to be run from the user venv, which would have `pyserial` installed as part of installing `esptool`.
Now as we vendor it inside the Mu source code we need to execute it in an environemnt that has `pyserial` installed (pyserial is no longer installed in the user venv), so we now have to use `sys.executable` which points to the python executable+environment that is running Mu, which will have `pyserial`.

Reply:
Of course. Thank you for the clarification.